tests/state_test.go

```diff
@@ -88,7 +88,7 @@ func TestState(t *testing.T) {
 				t.Run(key+"/snap", func(t *testing.T) {
 					withTrace(t, test.gasLimit(subtest), func(vmconfig vm.Config) error {
 						snaps, statedb, err := test.Run(subtest, vmconfig, true)
-						if snaps != nil && statedb != nil {
+						if err == nil && snaps != nil && statedb != nil {
```

Comment:
Butt why?

Reply:
If all transactions are rejected, the test.Run returns an error, and no snaps. And the `snaps.Journal` will then return a different error, obscuring the first one

Reply:
Should we perhaps delete `snaps != nil && statedb != nil` then? I just don't want to end up with skipping the snap clause altogether.

Reply:
Yes, that should be fine. Let's test it